app/models/pet.rb

```diff
@@ -65,6 +65,9 @@ class Pet < ApplicationRecord
     WEIGHT_UNIT_KG
   ]
 
+  scope :adopted, -> { Pet.includes(:match).where.not(match: {id: nil}) }
```

Comment:
Do we need to consider the tenant here? Not on my machine to test. I will check it out later unless you beat me to it.

Reply:
Update. All good.